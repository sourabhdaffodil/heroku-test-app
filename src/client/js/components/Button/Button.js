/*  eslint no-confusing-arrow: 0  */

import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { media } from '../../theme/style-utils';

const Btn = styled.button`
  border-radius: 3px;
  padding: 0;
  display: flex;
  ${props => props.small ? props.theme.textSetting('xs') : props.theme.textSetting('sm')}; // font size and line height
  font-weight: ${props => props.theme.weight('semiBold')};
  color: ${props => props.theme.contrast('light')};
  background-color: ${props => props.light ? props.theme.color('background', 'lighter') : '#eff3f6'};
  border: 1px solid ${props => props.theme.color('background')};
  cursor: pointer;
  line-height: 20px;

  ${media.phone`
    ${props => props.theme.textSetting('xs')}
  `}
`;

const Button = props => (
  <Btn {...props}>
    { props.children }
  </Btn>
);

Button.propTypes = {
  children: PropTypes.any.isRequired,
};

export default Button;
