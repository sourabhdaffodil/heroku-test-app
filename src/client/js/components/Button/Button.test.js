import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button component', () => {
  it('Must render a button', () => {
    const wrapper = shallow(
      <Button>
        My button
      </Button>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Must render a pill button', () => {
    const wrapper = shallow(
      <Button>
        <span>s</span>
      </Button>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
