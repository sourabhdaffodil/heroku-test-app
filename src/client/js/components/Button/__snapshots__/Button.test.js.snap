// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Must render a button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button>
    My button
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": "My button",
    },
    "ref": null,
    "rendered": "My button",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": "My button",
      },
      "ref": null,
      "rendered": "My button",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Button component Must render a pill button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button>
    <span>
      s
    </span>
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <span>
        s
      </span>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "s",
      },
      "ref": null,
      "rendered": "s",
      "type": "span",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <span>
          s
        </span>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "s",
        },
        "ref": null,
        "rendered": "s",
        "type": "span",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
